deploy-idsvr-azure-aks.sh

```diff
@@ -45,210 +43,258 @@ pre_requisites_check() {
   fi
 
   # Check for Azure resource group existence
-  RG_EXISTS=$(az group exists -n $resource_group) || true
+  RG_EXISTS=$(az group exists -n "$resource_group_name") || true
   if ! [[ "$RG_EXISTS" == "true" ]]; then
-      echo "Please make sure that the Azure resource group $resource_group exists in the subscription"
-      exit 1 
+    echo "Please make sure that the Azure resource group '$resource_group_name' defined in the 'infrastructure-config/infra-config.json' exists in the subscription."
+    exit 1
   fi
+
   # To avoid accidental commit of sensitive data to repositories
   cp ./hooks/pre-commit ./.git/hooks
 
-  echo -e "\n"
 }
 
-
-read_cluster_config_file() {
-  echo "Reading the configuration from cluster-config/aks-cluster-config.json .."
-  while read -r NAME; read -r VALUE; do
+read_infra_config_file() {
+  echo "Reading the configuration from infrastructure-config/infra-config.json .."
+  while
+    read -r NAME
+    read -r VALUE
+  do
     if [ -z "$NAME" ]; then break; fi
 
-  export "$NAME"="$VALUE" 
+    export "$NAME"="$VALUE"
+
+  done <<<"$(jq -rc '.[] | .[] | "\(.Name)\n\(.Value)"' "infrastructure-config/infra-config.json")"
+}
+
+fill_templates() {
+  envsubst <terraform-config/main.auto.tfvars.template >terraform-config/main.auto.tfvars
+  envsubst <idsvr-config/helm-values.yaml.template >idsvr-config/helm-values.yaml
+  envsubst <ingress-nginx-config/helm-values.yaml.template >ingress-nginx-config/helm-values.yaml
+}
+
+determine_aks_cluster_creation_type() {
```

Comment:
Currently the script allows to select Terraform despite Terraform not being installed and fails later with a "command not found" error which is not nice. I see different solutions:

1. only show the options that will work (i.e. check which tools that are installed) and print out some info, e.g. "Some options are not available. Make sure to install the tools."
2. If there's only one option available, select that automatically and print out some info, e.g. "This tool will run Azure CLI to deploy the AKS cluster as this is the only tool available. Do you want to continue?"
3. after selecting an option, check if tools are installed and print out some error if the machine does not meet the prerequisites, e.g. "You need to have Terraform installed to use that option". Then let the users decide to cancel or change their choices.
4. you could change the parameters of the script and let the users decide at the beginning what kind of option they want to use, e.g. --install --terraform

I personally prefer solution 3.